src/TSTransformer/nodes/transformOptionalChain.ts

```diff
@@ -257,10 +259,12 @@ function transformOptionalChainInner(
 			if (item.callOptional) {
 				if (item.kind === OptionalChainItemKind.PropertyCall) {
 					baseExpression = luau.property(convertToIndexableExpression(baseExpression), item.name);
-				} else {
+				} else if (item.kind === OptionalChainItemKind.ElementCall) {
+					const expType = state.typeChecker.getNonOptionalType(state.getType(item.expression.expression));
```

Comment:
This call is unnecessary - `addOneIfArrayType` already covers undefined as a possible type. Line 263old/267new seems like the only needed change.